lib/Backends/NNPI/NNPIOptions.cpp

```diff
@@ -96,6 +97,28 @@ llvm::StringMap<std::string> NNPIOptions::getSupportedOptions() {
   return supportedOptions_;
 }
 
+unsigned NNPIOptions::getFirstDeviceSteppingVersion() {
+  std::ifstream inFile;
+  constexpr char stepLoc[] = "/sys/class/nnpi/nnpi0/card_stepping";
+  inFile.open(stepLoc);
+  if (!inFile.good() || inFile.eof()) {
+    LOG(INFO) << "Could not find device stepping at " << stepLoc;
+    return 0;
+  }
+
+  // Only value in the file should be a single int for which step we're using.
+  std::string stepping;
+  getline(inFile, stepping);
+  inFile.close();
+
+  int devVer = NNPIOptions::getStringAsType<int>(stepping);
+  if (devVer < 0) {
+    // Not a valid stepping (must be a string with non negative integer).
+    return 0;
+  }
+  return devVer + 1;
```

Comment:
Please keep the comment from before to make clear what's going on:
```suggestion
  // Stepping is off by one vs. deviceVersion.
  return devVer + 1;
```

Reply:
Will fix.